Add Entry: HTML Comments (#614)
Codecademy/docs: content/html/concepts/comments/comments.md

---
Title: 'Comments'

> Description: 'HTML comments are markup that are included in the HTML source, but do not affect the rendering of the HTML document.'

Reviewer: ```suggestion
Description: 'HTML comments are markup included in the HTML source that do not affect the rendering of the HTML document.'
```

Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Comments'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

HTML comments are markup that are included in the HTML source, but do not affect the rendering of the HTML document. They are used for documenting the HTML code and for "commenting out" HTML that the developer doesn't want to render, but does not desire to delete, generally for historical or debugging purposes.

## Syntax

```html
<!-- comment goes here -->
```

> An HTML comment can contain any other HTML markup between the opening `<!--` and the closing `-->` except for another comment. Attempting to nest HTML comments will result in unexpected behavior because the comment is closed at the first `-->` it encounters. All text after the first `-->` will render, including the outer enclosing `-->`. Thus, when commenting out HTML code, the developer must make sure any interior comments are removed or strip the trailing `-->`s.

Reviewer: ```suggestion
An HTML comment can contain any other HTML markup between the opening `<!--` and the closing tag `-->` except for another comment. Attempting to nest HTML comments will result in unexpected behavior because the comment is closed at the first `-->` it encounters. All text after the first `-->` will render, including the outer enclosing `-->`. Thus, when commenting out HTML code, the developer must make sure any interior comments are removed or strip the trailing `-->`s.
```

> Also, an HTML comment can be placed anywhere inside another HTML element except inside the tags themselves. In other words they can't be placed between the opening `<` and closing `>` of a another tag.

Reviewer: ```suggestion
Also, an HTML comment can be placed anywhere inside another HTML element except inside the tags themselves. In other words, they can't be placed between the left angle `<` and right angle bracket`>` of a tag.
```

---
Title: 'Comments'

> Description: 'HTML comments are markup included in the HTML source that do not affect the rendering of the HTML document.'

Reviewer 2: ```suggestion
Description: 'HTML comments are used for documentation and excluding markup from being rendered.'
```

Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Comments'
  - 'Documentation'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

> HTML [comments](https://www.codecademy.com/resources/docs/general/comment) are markup included in the HTML source that do not affect the rendering of the HTML document. They are used for documenting the HTML code. They are also used for commenting out HTML that the developer doesn't want to render but does not desire to delete, either for historical or debugging purposes.

Reviewer 2: ```suggestion
HTML comments are used for documentation and excluding markup from being rendered. Like with [comments](https://www.codecademy.com/resources/docs/general/comment) in other languages, developers may also use HTML comments for historical or debugging purposes.
```

## Syntax

```html

> <!-- Comment goes here -->

Reviewer 2: ```suggestion
<!-- Comments go between these tags. -->
```

```

> An HTML comment can contain any other HTML markup between the opening `<!--` and the closing tag `-->` except for another comment. Attempting to nest HTML comments will result in unexpected behavior because the comment is closed at the first `-->` it encounters. All text after the first `-->` will render, including the outer enclosing `-->`. Thus, when commenting out HTML code, the developer must make sure any interior comments are removed or strip the trailing `-->`s.

Reviewer 2: ```suggestion
HTML comments always start with an opening `<!--` and end with a closing tag `-->`. They may contain anything from text to other markup that will not be rendered. 

Nested comments are not recommended because they may cause unexpected behavior. Comments generally end after the first closing tag `-->`. Anything after that will be rendered on the page, including all outer closing tags. Thus, when commenting out HTML code, developers should remove any interior comment tags (or at least the trailing ones `-->`).
```

> Also, an HTML comment can be placed anywhere inside another HTML element except inside the tags themselves. In other words, they can't be placed between the left angle `<` and right angle bracket`>` of a tag.

Reviewer 2: ```suggestion
Comment can also be placed anywhere inside another [element](https://www.codecademy.com/resources/docs/html/elements). However, they cannot be placed inside the element tags themselves (e.g. between the left angle `<` and right angle bracket`>` of a tag).
```

```pseudo
<p>Some text that <!-- This is a legal comment --> gets displayed</p>

<p <!-- This is an illegal comment -->>Some more text.</p>

<!-- Outer comment <!-- Inner comment --> This text will render. -->
```

## Example

```html

> <p>This will display.</p>

Reviewer 2: Should we include an image (in the `media/` folder) that shows what this rendered markup would look like?

## Syntax

HTML comments always start with an opening `<!--` and end with a closing tag `-->`. They may contain anything from text to other markup that will not be rendered.

```html
<!-- Comments go between these tags. -->
```

Nested comments are not recommended because they may cause unexpected behavior. Comments generally end after the first closing tag `-->`. Anything after that will be rendered on the page, including all outer closing tags. Thus, when commenting out HTML code, developers should remove any interior comment tags (or at least the trailing ones `-->`).

```pseudo
<!-- Outer comment <!-- Inner comment --> This text will render. -->
```

This example will render as follows:

![HTML Comments A](https://raw.githubusercontent.com/Codecademy/docs/main/media/html-comments-2.png)

Comments can also be placed anywhere inside another [element](https://www.codecademy.com/resources/docs/html/elements).

```pseudo
<p>Some text that <!-- This is a legal comment --> gets displayed</p>
```

> However, they cannot be placed inside the element tags themselves (e.g. between the left angle `<` and right angle bracket`>` of a tag).

Reviewer 3: ```suggestion
However, they cannot be placed inside the element tags themselves (e.g. between the left angle `<` and right angle bracket `>` of a tag).
```
